network/manager.go

```diff
@@ -517,9 +530,17 @@ func (nm *networkManager) GetEndpointInfo(networkID, endpointID string) (*Endpoi
 
 	if nm.IsStatelessCNIMode() {
 		logger.Info("calling cns getEndpoint API")
-		epInfo, err := nm.GetEndpointState(networkID, endpointID)
+		epInfos, err := nm.GetEndpointState(networkID, endpointID)
+		if err != nil {
+			return nil, err
+		}
+		for _, epInfo := range epInfos {
+			if epInfo.NICType == cns.InfraNIC {
+				return epInfo, nil
+			}
```

Comment:
why return only for infranic?

Reply:
get endpoint info's method signature allows me to only return one endpoint info, and when we get the endpoint state (stateless) there can be many eps associated with a single container id. sidenote: when do we even use GetEndpointInfo-- it uses "Get" flow but when do we use that??